sdk-lib/src/main/java/cash/z/ecc/android/sdk/SdkSynchronizer.kt

```diff
@@ -736,71 +748,78 @@ class SdkSynchronizer internal constructor(
- * @param txManager the component that manages outbound transactions in order to report which ones are
- * still pending, particularly after failed attempts or dropped connectivity. The intent is to help
- * monitor outbound transactions status through to completion.
- * @param processor the component responsible for processing compact blocks. This is effectively the
- * brains of the synchronizer that implements most of the high-level business logic and determines
- * the current state of the wallet.
+ * See the helper methods for generating default values.
  */
-@Suppress("FunctionName")
-fun Synchronizer(
-    initializer: Initializer,
-    repository: TransactionRepository =
-        PagedTransactionRepository(initializer.context, 1000, initializer.rustBackend, initializer.birthday, initializer.viewingKeys, initializer.overwriteVks), // TODO: fix this pagesize bug, small pages should not crash the app. It crashes with: Uncaught Exception: android.view.ViewRootImpl$CalledFromWrongThreadException: Only the original thread that created a view hierarchy can touch its views. and is probably related to FlowPagedList
-    blockStore: CompactBlockStore = CompactBlockDbStore(initializer.context, initializer.rustBackend.pathCacheDb),
-    service: LightWalletService = LightWalletGrpcService(initializer.context, initializer.host, initializer.port),
-    encoder: TransactionEncoder = WalletTransactionEncoder(initializer.rustBackend, repository),
-    downloader: CompactBlockDownloader = CompactBlockDownloader(service, blockStore),
-    txManager: OutboundTransactionManager =
-        PersistentTransactionManager(initializer.context, encoder, service),
-    processor: CompactBlockProcessor =
-        CompactBlockProcessor(downloader, repository, initializer.rustBackend, initializer.rustBackend.birthdayHeight)
-): Synchronizer {
-    // call the actual constructor now that all dependencies have been injected
-    // alternatively, this entire object graph can be supplied by Dagger
-    // This builder just makes that easier.
-    return SdkSynchronizer(
+object DefaultSynchronizerFactory {
+
+    fun new(
+        repository: TransactionRepository,
+        txManager: OutboundTransactionManager,
+        processor: CompactBlockProcessor,
+    ): Synchronizer {
+        // call the actual constructor now that all dependencies have been injected
+        // alternatively, this entire object graph can be supplied by Dagger
+        // This builder just makes that easier.
+        return SdkSynchronizer(
+            repository,
+            txManager,
+            processor
+        )
+    }
+
+    // TODO: fix this pagesize bug, small pages should not crash the app. It crashes with: Uncaught Exception: android.view.ViewRootImpl$CalledFromWrongThreadException: Only the original thread that created a view hierarchy can touch its views. and is probably related to FlowPagedList
```

Comment:
do we have a ticket for this issue?

Reply:
Yes.  This is a pre-existing todo comment; it just got moved around with this commit.  The issue is here https://github.com/zcash/zcash-android-wallet-sdk/issues/242

(Although that issue describes the problem, it may not be the implementation we want.)

I can add the issue number to this todo so that its more explicit.

Reply:
excellent!